cmd/grpcurl/grpcurl.go

```diff
@@ -509,17 +513,45 @@ is given then the descriptors for all exposed or known services are shown.
 
 If neither verb is present, the symbol must be a fully-qualified method name in
 'service/method' or 'service.method' format. In this case, the request body will
-be used to invoke the named method. If no body is given, an empty instance of
-the method's request type will be sent.
+be used to invoke the named method. If no body is given but one is required
+(i.e. the method is unary or server-streaming), an empty instance of the
+method's request type will be sent.
 
 The address will typically be in the form "host:port" where host can be an IP
 address or a hostname and port is a numeric port or service name. If an IPv6
 address is given, it must be surrounded by brackets, like "[2001:db8::1]". For
 Unix variants, if a -unix=true flag is present, then the address must be the
 path to the domain socket.
+
+Available flags:
 `, os.Args[0])
 	flag.PrintDefaults()
+}
 
+func prettify(docString string) string {
+	var buf bytes.Buffer
+	first := true
+	for {
+		pos := strings.IndexByte(docString, '\n')
+		if pos < 0 {
+			pos = len(docString)
+		}
+		line := strings.TrimSpace(docString[:pos])
+		if line != "" {
+			if first {
+				first = false
+			} else {
+				buf.WriteByte('\n')
+				buf.WriteString(indent())
+			}
+			buf.WriteString(line)
+		}
+		if pos >= len(docString) {
+			break
+		}
+		docString = docString[pos+1:]
+	}
+	return buf.String()
```

Comment:
would strings.Split not make this much more concise?

Reply:
Yes it would. Not sure why I formulated it this way.

Reply:
Fixed